ament_tools/verbs/build_pkg/cli.py

```diff
@@ -315,6 +315,12 @@ def run(opts, context):
     pkg_name = context.package_manifest.name
 
     if not opts.skip_build:
+        ignore_file = os.path.join(context.build_space, 'AMENT_IGNORE')
+        if not os.path.exists(ignore_file) and not context.dry_run:
+            os.makedirs(context.build_space, exist_ok=True)
+            with open(ignore_file, 'w'):
```

Comment:
We want `w` ("open for writing, truncating the file first") and not `+` ("open a disk file for updating (reading and writing)") (see https://docs.python.org/3/library/functions.html#open).